signature_body.md

```diff
@@ -0,0 +1,292 @@
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [keylime/enhancements] referencing this enhancement and targeting a release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Keylime i.e., [keylime/keylime], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+- [ ] Enhancement issue in release milestone, which links to pull request in [keylime/enhancements]
+- [ ] Core members have approved the issue with the label `in-progress`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place
+- [ ] User-facing documentation has been created in [keylime/keylime-docs]
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+## Summary
+
+This proposal introduces a method to sign over the body of an allow-list
+and then attach a signature and public key ID or X509 certificate.
+
+It uses the same manifest layout as supply chain projects [intoto](https://in-toto.io/)
+and [The update framework](https://theupdateframework.io/).
+<!--
+This section is incredibly important for producing high quality user-focused
+documentation such as release notes or a development roadmap.  It should be
+possible to collect this information before implementation begins in order to
+avoid requiring implementers to split their attention between writing release
+notes and implementing the feature itself. Reviewers
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+-->
+
+## Motivation
+
+The current implementation leverages seperate keys to the list itself. While
+suitable for some cases, it is also useful to have a single package where the
+list and the signature travel in the same vehicle. 
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this enhancement.  Describe why the change is important and the benefits to users.
+-->
+
+### Proposal
+
+User produces a cryptographic signature by 'passing over' the body of an allow/exclude
+list. The signature and public key id (sha256 hash of pubKey), or an X509 cert are then attached to the allowlist.
+
+The verifier upon discovery of a `"signed"` and `"signatures"` keys, will validate
+the signed section, using the keyid value and then pass the signed body to IMA.
+
+```json
+{
+    "signatures": [
+        {
+         "keyid": "{base64-encoded-x509-or-pub-key-digest}",
+         "sig": "{base64-encoded-signature}"
```

Comment:
Do we want to add a format identifier here?  E.g `"format": "GPGv1"` or  `"format": "X509v1"`

Reply:
Good catch, I had not added the keytype yet.